apps/system/js/acl_manager.js

```diff
@@ -0,0 +1,283 @@
+/* global dump, applications, AppWindowManager*/
+'use strict';
+
+(function(exports) {
+  var DEBUG = false;
+  var HOST = '127.0.0.1';
+  var PORT = 33334;
+  var RETRY_TIMEOUT = 1000;
+  var POLL_TIMEOUT = 60000 * 5; // 5 minutes
+  var SETTINGS_KEY = 'acl.enabled';
+  var NOTIFICATION_PREFIX = 'ACL_NOTIFICATION';
+
+  function debug() {
+    DEBUG && dump('[ACL] ' +
+      Array.prototype.slice.call(arguments).join(', ') + '\n');
+  }
+
+  function getRandomString() {
+    return Math.random().toString(36).replace(/[^a-z]+/g, '');
+  }
+
+  var ACLManager = function() {
```

Comment:
You're going to hate me for this, but I'm going to ask you to rename this. IMO `ACL` typically implies access control list in software. Using this abbreviation is confusing to me. I suggest: `app_compatibility.js`, and `var AppCompatibility = function() {`